petastorm/arrow_reader_worker.py

```diff
@@ -100,13 +100,17 @@ def __init__(self, worker_id, publish_func, args):
         self._transform_spec = args[6]
         self._transformed_schema = args[7]
         self._arrow_filters = args[8]
+        self._shuffle_rows = args[9]
+        self._shuffle_seed = args[10]
 
         if self._ngram:
             raise NotImplementedError('ngrams are not supported by ArrowReaderWorker')
 
         # We create datasets lazily in the first invocation of 'def process'. This speeds up startup time since
         # all Worker constructors are serialized
         self._dataset = None
+        if self._shuffle_seed:
+            np.random.seed(seed=self._shuffle_seed)
```

Comment:
This uses global state, right? This might create conflict with other users of the random generator, running on this or other threads.If we want to setup seed, I suggest we create a seeded random number generator and use it.
```python
random_state = np.random.RandomState(42)
random_state.permutations(10)
```

Reply:
Let me test. I was not sure if process pool needs to recreate a new random state.

Reply:
addressed.